sql/semisync_master.cc

```diff
@@ -826,6 +824,11 @@ int Repl_semi_sync_master::commit_trx(const char* trx_wait_binlog_name,
 
     set_timespec(start_ts, 0);
 
+    if (!m_wait_timeout)
```

Comment:
It is probably more consistent to extend the `if` at line 817 with ` && m_wait_timeout)`